api/internal/features/domain/controller/create_domain.go

```diff
@@ -42,6 +42,21 @@ func (c *DomainsController) CreateDomain(f fuego.ContextWithBody[types.CreateDom
 
 	if err != nil {
 		c.logger.Log(logger.Error, err.Error(), "")
+
+		if err == types.ErrDomainAlreadyExists {
+			return nil, fuego.HTTPError{
+				Err:    err,
+				Status: http.StatusConflict,
+			}
+		}
+
+		if err == types.ErrDomainNameInvalid || err == types.ErrDomainNameTooLong || err == types.ErrDomainNameTooShort || err == types.ErrMissingDomainName {
```

Comment:
Interchange the order of error message, Bad Request can come earlier than the Conflict